Doc/library/abc.rst

```diff
@@ -129,14 +130,14 @@ a helper class :class:`ABC` to alternatively define ABCs through inheritance:
                   yield None
 
           def get_iterator(self):
-              return self.__iter__()
+              return iter(self)
 
           @classmethod
           def __subclasshook__(cls, C):
-              if cls is MyIterable:
-                  if any("__iter__" in B.__dict__ for B in C.__mro__):
+              if cls is MyIterable:  # Do not inherit this behaviour
+                  if hasattr(C, "__iter__"):
```

Comment:
The current implementation is fast, clear, and correct. The new implementation here is slower, because it has to go through the long and complicated lookup that CPython does when `getattr()` is called. I don't find it any clearer. Lastly, the new implementation is more dangerous -- there's a possibility that arbitrary code could be executed if, for example, the metaclass has an `__iter__` property with side effects.

Reply:
Well, actually, both of these checks are incorrect :)

Counter example:

```python
from abc import *

class MyIterable(ABC):

        @abstractmethod
        def __iter__(self):
            while False:
                yield None

        @classmethod
        def __subclasshook__(cls, C):
            print(cls, C)
            if cls is MyIterable:  # Do not inherit this behaviour
                if any("__iter__" in B.__dict__ for B in C.__mro__):
                    print('Yes')
                    return True
            return NotImplemented  # fallback to regular subclass check


class My:
    __iter__ = None  # explictly not iterable

print(issubclass(My, MyIterable))  # True
```

I will change it to be `any(callable(B.__dict__.get("__iter__")) for B in C.__mro__):`
(This feels quite hard to read for me)

Reply:
Or `is not None` 👍